src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3432,6 +3476,789 @@ public void Inheritance_17()
+        }
+    }
+
+    static void Main()
+    {
+        M(new C(0));
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,14): error CS0102: The type 'C' already contains a definition for 'X'
+                // record C(int X)
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "X").WithArguments("C", "X").WithLocation(4, 14));
+
+            Assert.Equal(
+                "void C.Deconstruct(out System.Int32 X)",
+                comp.GetMember("C.Deconstruct").ToTestDisplayString(includeNonNullable: false));
+        }
+
+        [Fact]
+        public void Deconstruct_EventCollision()
+        {
+            var source = @"
+using System;
+
+record C(Action X)
+{
+    event Action X;
+
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C(Action x):
+                Console.Write(x);
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new C(() => { }));
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,17): error CS0102: The type 'C' already contains a definition for 'X'
+                // record C(Action X)
+                Diagnostic(ErrorCode.ERR_DuplicateNameInClass, "X").WithArguments("C", "X").WithLocation(4, 17));
+
+            Assert.Equal(
+                "void C.Deconstruct(out System.Action X)",
+                comp.GetMember("C.Deconstruct").ToTestDisplayString(includeNonNullable: false));
+        }
+
+        [Fact]
+        public void Deconstruct_Empty()
+        {
+            var source = @"
+record C
+{
+    static void M(C c)
+    {
+        switch (c)
+        {
+            case C():
+                break;
+        }
+    }
+
+    static void Main()
+    {
+        M(new C());
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,19): error CS1061: 'C' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'C' could be found (are you missing a using directive or an assembly reference?)
+                //             case C():
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "()").WithArguments("C", "Deconstruct").WithLocation(8, 19),
+                // (8,19): error CS8129: No suitable 'Deconstruct' instance or extension method was found for type 'C', with 0 out parameters and a void return type.
+                //             case C():
+                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "()").WithArguments("C", "0").WithLocation(8, 19));
+
+            Assert.Null(comp.GetMember("C.Deconstruct"));
+        }
+
+        [Fact]
+        public void Deconstruct_Inheritance_01()
+        {
+            var source = @"
+using System;
+
+record B(int X, int Y)
+{
+    internal B() : this(0, 1) { }
+}
+
+record C : B
```

Comment:
`record C(int X, int Y) : B`

Reply:
Followed up in https://github.com/dotnet/roslyn/pull/44806#discussion_r436987864